packages/amazonq/src/app/inline/recommendationService.ts

```diff
@@ -119,7 +122,51 @@ export class RecommendationService {
             })
             const t0 = performance.now()
 
-            const result = await this.getRecommendationsWithTimeout(languageClient, request, token)
+            // Best effort estimate of deletion
+            const isTriggerByDeletion = documentEventListener.isLastEventDeletion(document.uri.fsPath)
```

Comment:
@leigaol fyi, logic to handle trigger when users are deleting the code is moved here